From 8c112b6861883ebda66c4bbddd741f9e64f9293d Mon Sep 17 00:00:00 2001
From: spachatx <satish.chandrax.pachatu.prabhakara@intel.com>
Date: Thu, 7 Jan 2016 18:54:26 -0800
Subject: [PATCH 2/4] atomisp2: css2400 and 2401 irci_ecr-master_20160107_1930

Merge CSS2.1 for css2400 and 2401_legacy
for irci_ecr-master_20160107_1930 firmware tag

Change-Id: I2e74dfc47c565e63787c3598a673214b42322306
Signed-off-by: spachatx <satish.chandrax.pachatu.prabhakara@intel.com>
Tracked-On:
Reviewed-on: https://icggerrit.ir.intel.com/94085
Build: Dessert M BXT RVP (SC)
Reviewed-by: Smita Kulkarni <smita.kulkarni@intel.com>
Approver: WeiX Meng <weix.meng@intel.com>
Build: Dessert M CHT FFD (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-15952
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/458629
---
 .../css/css_2401_csi2p_system/csi_rx_global.h      | 32 +++++++++-----
 .../css/css_2401_csi2p_system/host/csi_rx.c        | 32 +++++++++-----
 .../css/css_2401_csi2p_system/host/csi_rx_local.h  | 32 +++++++++-----
 .../css_2401_csi2p_system/host/csi_rx_private.h    | 32 +++++++++-----
 .../css/css_2401_csi2p_system/host/ibuf_ctrl.c     | 32 +++++++++-----
 .../css_2401_csi2p_system/host/ibuf_ctrl_local.h   | 32 +++++++++-----
 .../css_2401_csi2p_system/host/ibuf_ctrl_private.h | 32 +++++++++-----
 .../host/input_system_local.h                      | 32 +++++++++-----
 .../host/input_system_private.h                    | 32 +++++++++-----
 .../css/css_2401_csi2p_system/host/isys_dma.c      | 32 +++++++++-----
 .../css_2401_csi2p_system/host/isys_dma_local.h    | 32 +++++++++-----
 .../css_2401_csi2p_system/host/isys_dma_private.h  | 32 +++++++++-----
 .../css/css_2401_csi2p_system/host/isys_irq.c      | 32 +++++++++-----
 .../css_2401_csi2p_system/host/isys_irq_local.h    | 32 +++++++++-----
 .../css_2401_csi2p_system/host/isys_irq_private.h  | 32 +++++++++-----
 .../css_2401_csi2p_system/host/isys_stream2mmio.c  | 32 +++++++++-----
 .../host/isys_stream2mmio_local.h                  | 32 +++++++++-----
 .../host/isys_stream2mmio_private.h                | 32 +++++++++-----
 .../css_2401_csi2p_system/host/pixelgen_local.h    | 32 +++++++++-----
 .../css_2401_csi2p_system/host/pixelgen_private.h  | 32 +++++++++-----
 .../css/css_2401_csi2p_system/host/system_local.h  | 32 +++++++++-----
 .../css/css_2401_csi2p_system/ibuf_ctrl_global.h   | 32 +++++++++-----
 .../css_2401_csi2p_system/input_system_global.h    | 32 +++++++++-----
 .../css/css_2401_csi2p_system/isys_dma_global.h    | 32 +++++++++-----
 .../css/css_2401_csi2p_system/isys_irq_global.h    | 32 +++++++++-----
 .../isys_stream2mmio_global.h                      | 32 +++++++++-----
 .../css/css_2401_csi2p_system/pixelgen_global.h    | 32 +++++++++-----
 .../css/css_2401_csi2p_system/system_global.h      | 32 +++++++++-----
 .../media/pci/atomisp2/css/css_api_version.h       |  8 +++-
 .../camera/drivers/media/pci/atomisp2/css/ia_css.h |  2 +-
 .../drivers/media/pci/atomisp2/css/ia_css_types.h  |  2 +-
 .../media/pci/atomisp2/css/ia_css_version_data.h   |  2 +-
 .../kernels/eed1_8/ia_css_eed1_8_default.host.c    |  4 +-
 .../css/isp/kernels/eed1_8/ia_css_eed1_8_types.h   |  6 +--
 .../atomisp2/css/isp/modes/interface/isp_const.h   |  3 +-
 .../css/runtime/binary/interface/ia_css_binary.h   |  2 +-
 .../camera/drivers/media/pci/atomisp2/css/sh_css.c | 49 +++++++++++++++++-----
 .../media/pci/atomisp2/css/sh_css_firmware.c       |  4 +-
 38 files changed, 674 insertions(+), 304 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/csi_rx_global.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/csi_rx_global.h
index 006ab1e..c9e76c6 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/csi_rx_global.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/csi_rx_global.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __CSI_RX_GLOBAL_H_INCLUDED__
 #define __CSI_RX_GLOBAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx.c
index 8f6ed1c..233e949 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx.c
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #include <type_support.h>
 #include "system_global.h"
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx_local.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx_local.h
index 6409e56..0ee3e23 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx_local.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx_local.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __CSI_RX_LOCAL_H_INCLUDED__
 #define __CSI_RX_LOCAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx_private.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx_private.h
index 25b7a93..280edc4e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx_private.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/csi_rx_private.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __CSI_RX_PRIVATE_H_INCLUDED__
 #define __CSI_RX_PRIVATE_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl.c
index 6844a3c..41c755d 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl.c
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #include <type_support.h>
 #include "system_global.h"
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl_local.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl_local.h
index 68989aa..53d7d54 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl_local.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl_local.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __IBUF_CTRL_LOCAL_H_INCLUDED__
 #define __IBUF_CTRL_LOCAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl_private.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl_private.h
index cce20b3..31acca6 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl_private.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/ibuf_ctrl_private.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __IBUF_CTRL_PRIVATE_H_INCLUDED__
 #define __IBUF_CTRL_PRIVATE_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/input_system_local.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/input_system_local.h
index 48a3e9b..b0ff72b 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/input_system_local.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/input_system_local.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __INPUT_SYSTEM_LOCAL_H_INCLUDED__
 #define __INPUT_SYSTEM_LOCAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/input_system_private.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/input_system_private.h
index 32a374c..4bf5425 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/input_system_private.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/input_system_private.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __INPUT_SYSTEM_PRIVATE_H_INCLUDED__
 #define __INPUT_SYSTEM_PRIVATE_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma.c
index 1f180e8..a747443 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma.c
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #include "isys_dma.h"
 #include "assert_support.h"
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma_local.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma_local.h
index 991d1da..01cba49 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma_local.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma_local.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __ISYS_DMA_LOCAL_H_INCLUDED__
 #define __ISYS_DMA_LOCAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma_private.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma_private.h
index 6ce5ef4..951bb22 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma_private.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_dma_private.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __ISYS_DMA_PRIVATE_H_INCLUDED__
 #define __ISYS_DMA_PRIVATE_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq.c
index 5c513e7..68c7f50 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq.c
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #include <system_local.h>
 #include "device_access.h"
 #include "assert_support.h"
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq_local.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq_local.h
index b7c80bd..26048bd 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq_local.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq_local.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __ISYS_IRQ_LOCAL_H__
 #define __ISYS_IRQ_LOCAL_H__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq_private.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq_private.h
index 129eeb2..e3e6228 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq_private.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_irq_private.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __ISYS_IRQ_PRIVATE_H__
 #define __ISYS_IRQ_PRIVATE_H__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio.c
index 638b109..2ee6abb 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio.c
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #include "isys_stream2mmio.h"
 
 const stream2mmio_sid_ID_t N_STREAM2MMIO_SID_PROCS[N_STREAM2MMIO_ID] = {
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio_local.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio_local.h
index 79515dd..1830c5d 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio_local.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio_local.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __ISYS_STREAM2MMIO_LOCAL_H_INCLUDED__
 #define __ISYS_STREAM2MMIO_LOCAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio_private.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio_private.h
index 60d19c3..8652e9c 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio_private.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/isys_stream2mmio_private.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __ISYS_STREAM2MMIO_PRIVATE_H_INCLUDED__
 #define __ISYS_STREAM2MMIO_PRIVATE_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/pixelgen_local.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/pixelgen_local.h
index 8680686..a368c2a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/pixelgen_local.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/pixelgen_local.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __PIXELGEN_LOCAL_H_INCLUDED__
 #define __PIXELGEN_LOCAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/pixelgen_private.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/pixelgen_private.h
index 4b46c08..1de976f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/pixelgen_private.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/pixelgen_private.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __PIXELGEN_PRIVATE_H_INCLUDED__
 #define __PIXELGEN_PRIVATE_H_INCLUDED__
 #include "pixelgen_public.h"
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/system_local.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/system_local.h
index ca8e359..ae2f4e8 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/system_local.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/host/system_local.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __SYSTEM_LOCAL_H_INCLUDED__
 #define __SYSTEM_LOCAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/ibuf_ctrl_global.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/ibuf_ctrl_global.h
index 2dd29df..547d3af 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/ibuf_ctrl_global.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/ibuf_ctrl_global.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __IBUF_CTRL_GLOBAL_H_INCLUDED__
 #define __IBUF_CTRL_GLOBAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/input_system_global.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/input_system_global.h
index 1ecd0d7..2a5b7d5 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/input_system_global.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/input_system_global.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __INPUT_SYSTEM_GLOBAL_H_INCLUDED__
 #define __INPUT_SYSTEM_GLOBAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_dma_global.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_dma_global.h
index 190628b..c5f041a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_dma_global.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_dma_global.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __ISYS_DMA_GLOBAL_H_INCLUDED__
 #define __ISYS_DMA_GLOBAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_irq_global.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_irq_global.h
index 142bc90..f9d831e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_irq_global.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_irq_global.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __ISYS_IRQ_GLOBAL_H__
 #define __ISYS_IRQ_GLOBAL_H__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_stream2mmio_global.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_stream2mmio_global.h
index d2dc961..99f1b02 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_stream2mmio_global.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/isys_stream2mmio_global.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __ISYS_STREAM2MMIO_GLOBAL_H_INCLUDED__
 #define __ISYS_STREAM2MMIO_GLOBAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/pixelgen_global.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/pixelgen_global.h
index 148dccf..8c22f6c 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/pixelgen_global.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/pixelgen_global.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __PIXELGEN_GLOBAL_H_INCLUDED__
 #define __PIXELGEN_GLOBAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/system_global.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/system_global.h
index 8b3c0ae..4b79ec3 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/system_global.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/system_global.h
@@ -1,16 +1,28 @@
 /*
-Support for Intel Camera Imaging ISP subsystem.
-Copyright (c) 2010 - 2015, Intel Corporation.
+* INTEL CONFIDENTIAL
+*
+* Copyright (C) 2014 - 2015 Intel Corporation.
+* All Rights Reserved.
+*
+* The source code contained or described herein and all documents
+* related to the source code ("Material") are owned by Intel Corporation
+* or licensors. Title to the Material remains with Intel
+* Corporation or its licensors. The Material contains trade
+* secrets and proprietary and confidential information of Intel or its
+* licensors. The Material is protected by worldwide copyright
+* and trade secret laws and treaty provisions. No part of the Material may
+* be used, copied, reproduced, modified, published, uploaded, posted,
+* transmitted, distributed, or disclosed in any way without Intel's prior
+* express written permission.
+*
+* No License under any patent, copyright, trade secret or other intellectual
+* property right is granted to or conferred upon you by disclosure or
+* delivery of the Materials, either expressly, by implication, inducement,
+* estoppel or otherwise. Any license under such intellectual property rights
+* must be express and approved by Intel in writing.
+*/
 
-This program is free software; you can redistribute it and/or modify it
-under the terms and conditions of the GNU General Public License,
-version 2, as published by the Free Software Foundation.
 
-This program is distributed in the hope it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-more details.
-*/
 #ifndef __SYSTEM_GLOBAL_H_INCLUDED__
 #define __SYSTEM_GLOBAL_H_INCLUDED__
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h
index 3712bc8..672127f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h
@@ -34,7 +34,7 @@ The version string has four dot-separated numbers, read left to right:
 
 */
 
-#define CSS_API_VERSION_STRING	"2.1.20.20"
+#define CSS_API_VERSION_STRING	"2.1.21.0"
 
 /*
 Change log
@@ -466,6 +466,12 @@ v2.1.20.19, Add Plane IO parameters in ISP configurations
 v2.1.20.20, Introduced YCgCo444_16 internal frame format
 - Replace ISP2.7 internal format V444_16 with correct YCgCo444_16
 
+v2.1.20.21, insert copyright header on CSS-API files
+- add copyright header
+
+v2.1.21.0, Reorder EED1_8 kernel parameters in CSS API
+- Reorder EED1_8 fields in ia_css_eed1_8_config
+
 */
 
 #endif /*__CSS_API_VERSION_H*/
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
index 8594fa4f..da8cc2a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20151229_1822 */
+/* Release Version: irci_ecr-master_20160107_1930 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
index 5a0b70c..6753eea 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20151229_1822 */
+/* Release Version: irci_ecr-master_20160107_1930 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
index 47d6bb2..46fbe2e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
@@ -22,7 +22,7 @@ more details.
 #define __IA_CSS_VERSION_DATA_H
 
 
-#define CSS_VERSION_STRING "REL:20151229_53.2_2234; API:2.1.20.20; GIT:irci____3#c6a33634a2caf2b12d3807c2438571e1790b0cd3; SDK:; USER:viedifw; "
+#define CSS_VERSION_STRING "REL:20160107_01.4_2023; API:2.1.21.0; GIT:irci___#428a21e7f937a777e612a8eaf5fec41478e09cd1; SDK:; USER:viedifw; "
 
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/eed1_8/ia_css_eed1_8_default.host.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/eed1_8/ia_css_eed1_8_default.host.c
index 90eb231..b60696a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/eed1_8/ia_css_eed1_8_default.host.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/eed1_8/ia_css_eed1_8_default.host.c
@@ -29,10 +29,10 @@ const struct ia_css_eed1_8_config default_eed1_8_config = {
 	.fc_coring_prm = 128,
 	.aerel_thres0 = 0,
 	.aerel_gain0 = 8191,
-	.aerel_thres1 = 16,
-	.aerel_gain1 = 20,
 	.derel_thres0 = 1229,
 	.derel_gain0 = 1,
+	.aerel_thres1 = 16,
+	.aerel_gain1 = 20,
 	.derel_thres1 = 819,
 	.derel_gain1 = 1,
 	.coring_pos0 = 0,
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/eed1_8/ia_css_eed1_8_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/eed1_8/ia_css_eed1_8_types.h
index 8fbc3d5..794f70e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/eed1_8/ia_css_eed1_8_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/eed1_8/ia_css_eed1_8_types.h
@@ -52,11 +52,11 @@ struct ia_css_eed1_8_config {
 
 	int32_t aerel_thres0;	/**< Threshold for Non-Directional Reliability at dark region. */
 	int32_t aerel_gain0;	/**< Gain for Non-Directional Reliability at dark region. */
-	int32_t aerel_thres1;	/**< Threshold for Non-Directional Reliability at bright region. */
-	int32_t aerel_gain1;	/**< Gain for Non-Directional Reliability at bright region. */
-
 	int32_t derel_thres0;	/**< Threshold for Directional Reliability at dark region. */
 	int32_t derel_gain0;	/**< Gain for Directional Reliability at dark region. */
+
+	int32_t aerel_thres1;	/**< Threshold for Non-Directional Reliability at bright region. */
+	int32_t aerel_gain1;	/**< Gain for Non-Directional Reliability at bright region. */
 	int32_t derel_thres1;	/**< Threshold for Directional Reliability at bright region. */
 	int32_t derel_gain1;	/**< Gain for Directional Reliability at bright region. */
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/modes/interface/isp_const.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/modes/interface/isp_const.h
index 0cc677c..0a20b1f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/modes/interface/isp_const.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/modes/interface/isp_const.h
@@ -217,7 +217,8 @@ more details.
 #define SH_CSS_BINARY_ID_PRIMARY_ISP27_STAGE0	            503
 #define SH_CSS_BINARY_ID_PRIMARY_ISP27_STAGE1		    504
 #define SH_CSS_BINARY_ID_PRIMARY_ISP27_STAGE2		    505
-#define SH_CSS_BINARY_ID_CAPTURE_PP_ISP27		    506
+#define SH_CSS_BINARY_ID_PRIMARY_ISP27_STAGE3		    506
+#define SH_CSS_BINARY_ID_CAPTURE_PP_ISP27		    507
 
 #define XMEM_WIDTH_BITS              HIVE_ISP_DDR_WORD_BITS
 #define XMEM_SHORTS_PER_WORD         (HIVE_ISP_DDR_WORD_BITS/16)
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h
index 8fbe479..b8e5405 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h
@@ -47,7 +47,7 @@ more details.
 
 #define MAX_NUM_PRIMARY_STAGES 6
 #define NUM_PRIMARY_HQ_STAGES  6  /* number of primary stages for ISP2.6.1 high quality pipe */
-#define NUM_PRIMARY_HQ27_STAGES 3 /* number of primary stages for ISP2.7 high quality pipe */
+#define NUM_PRIMARY_HQ27_STAGES 4 /* number of primary stages for ISP2.7 high quality pipe */
 #define NUM_PRIMARY_STAGES     1  /* number of primary satges for ISP1/ISP2.2 pipe */
 
 /* Indicate where binaries can read input from */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css.c
index 5271ddd..b7a5808 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css.c
@@ -160,6 +160,7 @@ static bool fw_explicitly_loaded = false;
 static enum ia_css_frame_format primary_hq_stage_out_format[NUM_PRIMARY_HQ27_STAGES] = {
 	IA_CSS_FRAME_FORMAT_YCgCo444_16,
 	IA_CSS_FRAME_FORMAT_YUV420_16,
+	IA_CSS_FRAME_FORMAT_YUV420,
 	IA_CSS_FRAME_FORMAT_YUV420
 };
 
@@ -6204,21 +6205,22 @@ static enum ia_css_err load_primary_binaries(
 
 	if (need_pre_de) {
 		struct ia_css_frame_info pre_de_in_info;
+		struct ia_css_frame_info pre_de_out_info;
 		struct ia_css_binary_descr pre_de_descr;
 		if (!pipe->pipe_settings.capture.pre_isp_binary.info) {
-			enum ia_css_frame_format out_format = pipe_out_info->format;
-			unsigned int out_pad_width = pipe_out_info->padded_width;
 
 			/* Temporary assign RAW output format for pre_de lookup.
 			 * It will be restored at the end of this block.
 			 */
-			pipe_out_info->format = IA_CSS_FRAME_FORMAT_RAW;
-			ia_css_frame_info_set_width(pipe_out_info,
-						pipe_out_info->res.width, 0);
-
+			pre_de_out_info = *pipe_out_info;
+			pre_de_out_info.format = IA_CSS_FRAME_FORMAT_RAW;
+			pre_de_out_info.res.width = pipe->config.input_effective_res.width;
+			pre_de_out_info.res.height = pipe->config.input_effective_res.height;
+			ia_css_frame_info_set_width(&pre_de_out_info,
+						pipe->config.input_effective_res.width, 0);
 			ia_css_pipe_get_pre_de_binarydesc(pipe, &pre_de_descr,
 						&pre_de_in_info,
-						pipe_out_info);
+						&pre_de_out_info);
 
 			err = ia_css_binary_find(&pre_de_descr,
 						&pipe->pipe_settings.capture.pre_isp_binary);
@@ -6227,9 +6229,6 @@ static enum ia_css_err load_primary_binaries(
 				IA_CSS_LEAVE_ERR_PRIVATE(err);
 				return err;
 			}
-
-			pipe_out_info->format = out_format;
-			pipe_out_info->padded_width = out_pad_width;
 		}
 	}
 
@@ -6314,6 +6313,7 @@ static enum ia_css_err load_primary_binaries(
 				&cas_scaler_descr.out_info[i],
 				&cas_scaler_descr.internal_out_info[i],
 				&cas_scaler_descr.vf_info[i]);
+
 #if defined(HAS_RES_MGR)
 			bds_out_info.res = pipe->config.bayer_ds_out_res;
 			yuv_scaler_descr.bds_out_info = &bds_out_info;
@@ -6374,6 +6374,8 @@ static enum ia_css_err load_primary_binaries(
 
 		for (i = 0; i < mycs->num_primary_stage; i++) {
 			struct ia_css_frame_info *local_vf_info = NULL;
+			struct ia_css_frame_info prim_stage_out_info = prim_out_info;
+
 			if (pipe->enable_viewfinder[IA_CSS_PIPE_OUTPUT_STAGE_0] && (i == mycs->num_primary_stage - 1))
 				local_vf_info = &vf_info;
 
@@ -6384,9 +6386,34 @@ static enum ia_css_err load_primary_binaries(
 				prim_out_info.format = primary_hq_stage_out_format[i];
 				ia_css_frame_info_set_width(&prim_out_info,
 					prim_out_info.res.width, 0);
+
+				if (i < (mycs->num_primary_stage - 1)) {
+					/* primary sub-stages (except for the last)
+					 * input == output
+					 */
+					prim_stage_out_info = prim_out_info;
+					prim_stage_out_info.res = pipe->config.input_effective_res;
+				} else {
+					/* primary last sub-stage
+					 * Crops fixed amount (128x96 pixels) from effective.
+					 * The output of the primary stage should therefore
+					 * be 'crop fixed res' smaller than effective.
+					 *
+					 * The output resolution of the primary stages
+					 * should be: 'capture_pp_in_res'.  If this is
+					 * not defined, it will be the output resolution
+					 * of the pipe.
+					 *
+					 * Horizontally: one vector from left and right
+					 * Vertically: 48 pixels from top and bottom
+					 */
+					prim_stage_out_info = prim_out_info;
+				}
+				ia_css_frame_info_set_width(&prim_stage_out_info,
+					prim_stage_out_info.res.width, 0);
 			}
 
-			ia_css_pipe_get_primary_binarydesc(pipe, &prim_descr[i], &prim_in_info, &prim_out_info, local_vf_info, i);
+			ia_css_pipe_get_primary_binarydesc(pipe, &prim_descr[i], &prim_in_info, &prim_stage_out_info, local_vf_info, i);
 #if defined(HAS_RES_MGR)
 			bds_out_info.res = pipe->config.bayer_ds_out_res;
 			prim_descr[i].bds_out_info = &bds_out_info;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
index c616dde..2e4c191 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
@@ -47,10 +47,10 @@ struct fw_param {
 /* Warning: same order as SH_CSS_BINARY_ID_* */
 static struct firmware_header *firmware_header;
 
-/* The string STR(irci_ecr-master_20151229_1822) is a place holder
+/* The string STR(irci_ecr-master_20160107_1930) is a place holder
  * which will be replaced with the actual RELEASE_VERSION
  * during package generation. Please do not modify  */
-static const char *release_version = STR(irci_ecr-master_20151229_1822);
+static const char *release_version = STR(irci_ecr-master_20160107_1930);
 
 #define MAX_FW_REL_VER_NAME	300
 static char FW_rel_ver_name[MAX_FW_REL_VER_NAME] = "---";
-- 
1.9.1

